test/test_nlp_expr.jl

```diff
@@ -1092,4 +1060,55 @@ function test_error_nonlinear_expr_complex_constructor()
     return
 end
 
+function test_error_legacy_expression_constructor()
+    model = Model()
+    @variable(model, x)
+    @NLexpression(model, arg, x^3)
+    err = ErrorException(
+        """
+        Cannot mix a legacy NonlinearExpression with the new nonlinear API.
+
+        Got: $arg
+
+        To update, replace all calls to `@NLexpression` with `@expression`.
+        """,
+    )
+    @test_throws err @objective(model, Min, arg)
+    @test_throws err @constraint(model, arg <= 0)
+    @test_throws err @constraint(model, arg in MOI.LessThan(0.0))
+    @test_throws err moi_function(arg)
+    @test_throws err x * arg
+    @test_throws err arg * x
```

Comment:
There's technically two unrelated bugs here: something like `@objective(model, Min, arg)`, and `x * arg`. But now we uniformly throw the same error, which is hopefully informative for the user.